### ISSUE TITLE
RealmObject.isValid(...) will return true on a `null` object

### ISSUE THREAD

**Issue by olivierg13** [realm/realm-java#5865]
Hey folks,

I used the RealmObject.isValid() method recently and I noticed the implementation was returning true for `null` objects:
```
/**
     * Checks if the RealmObject is still valid to use i.e., the RealmObject hasn't been deleted nor has the
     * {@link io.realm.Realm} been closed. It will always return {@code true} for unmanaged objects.
     *
     * @param object RealmObject to check validity for.
     * @return {@code true} if the object is still accessible or an unmanaged object, {@code false} otherwise.
     */
    public static <E extends RealmModel> boolean isValid(E object) {
        if (object instanceof RealmObjectProxy) {
            RealmObjectProxy proxy = (RealmObjectProxy) object;
            Row row = proxy.realmGet$proxyState().getRow$realm();
            return row != null && row.isAttached();
        } else {
            return true;
        }
    }
```

As `object instanceof RealmObjectProxy` will be false for a `null` object, the `return true` block is reached.

Is it the expected behavior?
I'm resolving this by doing `(object != null && RealmObject.isValid(object))`, so there are a lot of repeats, and if I forget to add the null check we could end up with NPEs.

What do you think?

Thank you in advance.

Cheers,

Olivier

**Comment by cmelchior** [comment]
Hey @olivierg13 
This looks like an oversight on our parts, mostly because we forgot to take into account that people might send `null` to that method.

I agree that `null` returning `false` would be a much more intuitive behaviour.

**Comment by cmelchior** [comment]
Also, given the Javadoc ` @return {@code true} if the object is still accessible or an unmanaged object, {@code false} otherwise.`, `false` should be returned.

Note that we are using `package-info` with `@javax.annotation.ParametersAreNonnullByDefault` so if you have tools checking for that you should never accidentially parse `null` to the method, but obviously we should still do the correct thing if done anyway.

**Comment by cmelchior** [comment]
Fixed in #5871

**Comment by olivierg13** [comment]
Thank you @cmelchior :)
